### PR TITLE
fix: Clean the Mongo database before each test

### PR DESCRIPTION
**Pull request by chombymat**

### CODE REVIEW

File: tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java [hunk 1]
```diff
@@ -46,6 +51,51 @@ public SpringMongoSteps(ObjectSteps objects, SpringSteps spring) {
...
+                    })
+                    .collect(Collectors.toMap(
+                            Map.Entry::getKey,
+                            Map.Entry::getValue,
+                            (v1, v2) -> v1
+                    ));
+        }
+        if (entityClassByCollectionName == null) {
+            entityClassByCollectionName = mongoRepositoryByClass.keySet().stream()
+                    .map(type -> (Class<?>) type)
```
Comment (mehdi-elketroussi) [inline tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java:85]: I don't understand the purpose of those lines of code. In my opinion, only
`        if (autoclean) {
            mongoTemplates.forEach(template -> template.getDb().drop());
        }`  is enough for the feature.

File: tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java [hunk 2]
```diff
@@ -46,6 +51,51 @@ public SpringMongoSteps(ObjectSteps objects, SpringSteps spring) {
 
     @Before
     public void before() {
+        if (autoclean) {
```
Comment (mehdi-elketroussi) [inline tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java:54]: I think you can write a test for this part:
Create a custom step to run this 3 lines, write a test which insert some data, execute the custom step for deletion and check the database is empty.

**Review by mehdi-elketroussi — CHANGES_REQUESTED** [review]
Thanks for your contribution, but some changes are needed.

File: tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java (hunk 2 above)
Comment (chombymat) [reply tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java:51]: Done.

File: tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java (hunk 1 above)
Comment (chombymat) [reply tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java]: Yep, i don't remember why... it's fixed

**Review by mehdi-elketroussi — APPROVED** [review]

**Pull request merged**